### ISSUE TITLE
Validation DataLoader applies the training augmentation, corrupting model selection

### ISSUE THREAD

**Issue by abuzarmahmood** [pulakatlab/autoslide#118]
Part of #116.

## The defect

`create_dataloaders` passes the same `transform` object to both the training and
the validation DataLoader:

https://github.com/pulakatlab/autoslide/blob/main/autoslide/src/pipeline/model/data_preprocessing.py#L729-L753

```python
train_dl = torch.utils.data.DataLoader(
    AugmentedCustDat(train_img_paths, train_mask_paths, transform), ...)
val_dl = torch.utils.data.DataLoader(
    AugmentedCustDat(val_img_paths, val_mask_paths, transform), ...)
```

`transform` comes from `create_transforms()`, which is the full random
augmentation stack: `RandomHorizontalFlip`, `RandomVerticalFlip`,
`RandomRotation90`, `RandomShear`, `ColorJitter`, and
`ElasticTransform(alpha=3000, sigma=30)`.

## Why it matters

`train_model` selects the released checkpoint on validation loss:

```python
if val_epoch_loss < best_val_loss:
    best_val_loss = val_epoch_loss
    torch.save(model.state_dict(), best_model_path)
```

Because validation images are randomly flipped, colour-jittered and elastically
warped on every pass, `val_epoch_loss` is a noisy estimate of a *different*
distribution than the one the model is deployed on. Model selection is
therefore partly driven by which epoch happened to draw an easier set of random
augmentation parameters.

Compounding it: the augmentation is unseeded, so the validation loss curve is
not reproducible between runs.

## Suggested fix

```python
val_dl = torch.utils.data.DataLoader(
    AugmentedCustDat(val_img_paths, val_mask_paths, T.ToTensor()), ...)
```

Separately worth revisiting: `ElasticTransform(alpha=3000, sigma=30)` is about
60x torchvision's default `alpha=50`. On 706 px images that is a very large
displacement field, and it is applied to masks as well as images. Worth an
ablation to confirm it helps rather than destroying vessel morphology.
